share/wake/lib/core/boolean.wake

```diff
@@ -28,11 +30,8 @@ export data Boolean =
 # !True  = False
 # !False = True
 # ```
-export def !(x: Boolean): Boolean =
-    if x then
-        False
-    else
-        True
+export def !(x: Boolean): Boolean = if x then False else True
```

Comment:
I also tend to be (or at least I think I am) strict about only using `if __ then __ else __` on a single line for very simple expressions.

As for the formatting between the way it was (before the change here) and Sam's suggested formatting, I'm not sure I have a strong opinion. At least in F#, I bounce between the two. I even sometimes use:

```wake
if ___
then ___
else ___
```

🙈

Reply:
I do like having all three keywords on separate lines as well, unless the `then` and `else` bodies are indented, in which case putting `then` at the end of the `if` line makes sense.

Reply:
So we are somewhat limited in what we can determine about the expression. The easiest is total line width but that a constant for all code. 

In the short term I think we need to pick between "allow non-optimal single line if/then/else" or "never allow single line if/then/else". In the long term, we can try to dial in hyper focused rules but I want to avoid those kinds of rules until we hit MVP.

Re: multi-line if/then/else, I have a pretty strong preference for 
```
if ______ then
    ______ 
else
    ______ 
```
over
```
if _____
then ______ 
else ______ 
```
lol

Reply:
I think a short ternary `if then else` is ok. When it's longer, I agree with @V-FEXrt 's multi-line format.